tests/output.rs

```diff
@@ -0,0 +1,176 @@
+#![allow(clippy::into_iter_on_ref)]
```

Comment:
I disagree stylistically with this lint, I think using `.into_iter()` on a reference type is fine and it is a stunted mental model to think that `.into_iter()` means "it moves the contents of the collection into the iterator" (if that's what the lint [docs](https://rust-lang.github.io/rust-clippy/master/index.html#/into_iter_on_ref) are indeed telling me). No, a reference type obviously doesn't own the contents of the collection, and it has a different `.into_iter()` implementation. Encouraging people to "conflate" `.into_iter()` with "owning" seems like a terrible idea

Also I think it's insane that there's no such thing as eg a workspace-wide config file for configuring individual Clippy lints. Like here I'd have to add this separately to `src/lib.rs` if I wanted it to "kick in" there